File: PST/FUEL/src/FUEL/clsSecurePropertyKey.vb
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Diagnostics
Imports System.Runtime.CompilerServices

Namespace FUEL
    Public Class clsSecurePropertyKey
        ' Methods
        Protected Friend Sub New()
        End Sub

        Public Sub New(ByVal Key As String)
            Me._DecryptedKey = New Simple3Des("Overlanding123").DecryptData(Key)
            Me._KeyExpirationDate = Me.GetDateFromDecryptedKey(Me._DecryptedKey)
        End Sub

        Private Function GetDateFromDecryptedKey(ByVal strKey As Object) As DateTime
            If Not Versioned.IsNumeric(strKey) Then
                Throw New ArgumentException("Malformed Key")
            End If
            Return New DateTime(Conversions.ToLong(Operators.DivideObject(strKey, 3.1415926535897931)))
        End Function


        ' Properties
        Private Property _DecryptedKey As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__DecryptedKey
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__DecryptedKey = AutoPropertyValue
            End Set
        End Property

        Private Property _KeyExpirationDate As DateTime
            <DebuggerNonUserCode> _
            Get
                Return Me.__KeyExpirationDate
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As DateTime)
                Me.__KeyExpirationDate = AutoPropertyValue
            End Set
        End Property

        Public ReadOnly Property KeyExpirationDate As DateTime
            Get
                Return Me._KeyExpirationDate
            End Get
        End Property

        Private Property _Key As String
            <DebuggerNonUserCode> _
            Get
                Return Me.__Key
            End Get
            <DebuggerNonUserCode> _
            Set(ByVal AutoPropertyValue As String)
                Me.__Key = AutoPropertyValue
            End Set
        End Property

        Public ReadOnly Property Key As String
            Get
                Return Me._Key
            End Get
        End Property

        Public ReadOnly Property KeyIsValid As Boolean
            Get
                Dim flag As Boolean
                If Not ((DateTime.Compare(Me._KeyExpirationDate.Date, DateAndTime.Now.Date) >= 0) Or (DateTime.Compare(Me._KeyExpirationDate.Date, Conversions.ToDate("06/06/2006").Date) = 0)) Then
                    flag = False
                Else
                    If (DateTime.Compare(Me._KeyExpirationDate, Conversions.ToDate("06/06/2006")) = 0) Then
                        Interaction.MsgBox("666", MsgBoxStyle.ApplicationModal, Nothing)
                    End If
                    flag = True
                End If
                Return flag
            End Get
        End Property


        ' Fields
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __DecryptedKey As String
        <CompilerGenerated, DebuggerBrowsable(DebuggerBrowsableState.Never)> _
        Private __KeyExpirationDate As DateTime
        <DebuggerBrowsable(DebuggerBrowsableState.Never), CompilerGenerated> _
        Private __Key As String
    End Class
End Namespace

File: PST/FUEL Source Code/FUEL/clsSecurePropertyKey.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.clsSecurePropertyKey
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Diagnostics;
using System.Runtime.CompilerServices;

namespace FUEL
{
  public class clsSecurePropertyKey
  {
    private string _DecryptedKey { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    private DateTime _KeyExpirationDate { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public DateTime KeyExpirationDate => this._KeyExpirationDate;

    private string _Key { [DebuggerNonUserCode] get; [DebuggerNonUserCode] set; }

    public string Key => this._Key;

    public bool KeyIsValid
    {
      get
      {
        if (!(DateTime.Compare(this._KeyExpirationDate.Date, DateAndTime.Now.Date) >= 0 | DateTime.Compare(this._KeyExpirationDate.Date, Conversions.ToDate("06/06/2006").Date) == 0))
          return false;
        if (DateTime.Compare(this._KeyExpirationDate, Conversions.ToDate("06/06/2006")) == 0)
        {
          int num = (int) Interaction.MsgBox((object) "666");
        }
        return true;
      }
    }

    private DateTime GetDateFromDecryptedKey(object strKey) => Versioned.IsNumeric(RuntimeHelpers.GetObjectValue(strKey)) ? new DateTime(Conversions.ToLong(Operators.DivideObject(strKey, (object) Math.PI))) : throw new ArgumentException("Malformed Key");

    protected internal clsSecurePropertyKey()
    {
    }

    public clsSecurePropertyKey(string Key)
    {
      this._DecryptedKey = new Simple3Des("Overlanding123").DecryptData(Key);
      this._KeyExpirationDate = this.GetDateFromDecryptedKey((object) this._DecryptedKey);
    }
  }
}
